p2p/protocol/identify/id.go

```diff
@@ -116,55 +129,92 @@ func NewIDService(ctx context.Context, h host.Host, opts ...Option) *IDService {
 
 	// handle local protocol handler updates, and push deltas to peers.
 	var err error
-	s.subscription, err = h.EventBus().Subscribe(&event.EvtLocalProtocolsUpdated{}, eventbus.BufSize(128))
+	s.subscriptions.localProtocolsUpdated, err = h.EventBus().Subscribe(&event.EvtLocalProtocolsUpdated{}, eventbus.BufSize(128))
+	if err != nil {
+		log.Warnf("identify service not subscribed to local protocol handlers updates; err: %s", err)
+	} else {
+		go s.handleEvents(s.subscriptions.localProtocolsUpdated, s.handleProtosChanged)
+	}
+	s.subscriptions.localPeerRecordUpdated, err = h.EventBus().Subscribe(&event.EvtLocalPeerRecordUpdated{}, eventbus.BufSize(128))
 	if err != nil {
-		log.Warningf("identify service not subscribed to local protocol handlers updates; err: %s", err)
+		log.Warnf("identify service not subscribed to local peer record changes; err: %s", err)
 	} else {
-		go s.handleEvents()
+		go s.handleEvents(s.subscriptions.localPeerRecordUpdated, s.handlePeerRecordUpdated)
+	}
+	s.subscriptions.localAddrsUpdated, err = h.EventBus().Subscribe(&event.EvtLocalAddressesUpdated{}, eventbus.BufSize(128))
+	if err != nil {
+		log.Warnf("identify service not subscribed to address changes. err: %s", err)
+	} else {
+		go s.handleEvents(s.subscriptions.localAddrsUpdated, s.handleLocalAddrsUpdated)
```

Comment:
@yusefnapora Wont this cause us to send a `Push` twice since we also do a Push when we get a `EvtLocalPeerRecordUpdated ` which in turn is also derived from `EvtLocalAddressesUpdated `?

Reply:
Nevermind. This has already been raised in https://github.com/libp2p/go-libp2p/pull/747#issuecomment-596537472